--- a/arch/x86/Kconfig
+++ b/arch/x86/Kconfig
@@ -245,7 +245,6 @@ config X86
 	select HAVE_NOINSTR_HACK		if HAVE_OBJTOOL
 	select HAVE_NMI
 	select HAVE_NOINSTR_VALIDATION		if HAVE_OBJTOOL
-	select HAVE_OBJTOOL			if X86_64
 	select HAVE_OPTPROBES
 	select HAVE_PCSPKR_PLATFORM
 	select HAVE_PERF_EVENTS
--- a/arch/x86/include/asm/unwind_hints.h
+++ b/arch/x86/include/asm/unwind_hints.h
@@ -16,7 +16,6 @@
 .endm
 
 .macro UNWIND_HINT_ENTRY
-	VALIDATE_UNRET_BEGIN
 	UNWIND_HINT_END_OF_STACK
 .endm
 
@@ -58,7 +57,6 @@
 .endm
 
 .macro UNWIND_HINT_IRET_ENTRY base=%rsp offset=0 signal=1
-	VALIDATE_UNRET_BEGIN
 	UNWIND_HINT_IRET_REGS base=\base offset=\offset signal=\signal
 .endm

--- a/arch/x86/entry/entry_64.S
+++ b/arch/x86/entry/entry_64.S
@@ -1107,7 +1107,6 @@ SYM_CODE_START(error_entry)
 	FENCE_SWAPGS_KERNEL_ENTRY
 	CALL_DEPTH_ACCOUNT
 	leaq	8(%rsp), %rax			/* return pt_regs pointer */
-	VALIDATE_UNRET_END
 	RET
 
 .Lbstep_iret:
